Lib/test/test_unittest/testmock/testhelpers.py

```diff
@@ -1107,6 +1107,32 @@ class WithNonFields:
                 with self.assertRaisesRegex(AttributeError, msg):
                     mock.b
 
+    def test_dataclass_with_wider_default(self):
```

Comment:
The default value type here is actually narrower than the field annotation , so describing it as `with_wider_default` ends up being confusing because "default" is being used in two different senses. (the new dataclass handling picks up the wider field annotation by default, so we want the narrower class attribute handling that reads the default value to override that)

Suggested names below:

```suggestion
    def test_dataclass_default_value_type_overrides_field_annotation(self):
```